client-app/src/Components/NewItemForm.tsx

```diff
@@ -15,35 +17,87 @@ interface FormErrors {
         currentStatus: 'Received',
         donorEmail: '',
+        donorId: null, // Initialize with null
         program: '',
+        programId: null, // Initialize with null
         imageUpload: [],
         dateDonated: '',
     });
 
-    const itemTypeOptions = [
-        { value: 'bicycle', label: 'Bicycle' },
-        { value: 'computer', label: 'Computer' },
-        // More item type options can be added here
-    ];
+    // const itemTypeOptions = [
+    //     { value: 'bicycle', label: 'Bicycle' },
+    //     { value: 'computer', label: 'Computer' },
+    //     // More item type options can be added here
+    // ];
+
+    // const donorEmailOptions = [
+    //     { value: 'email1', label: 'cooldude@gmail.com' },
+    //     { value: 'email2', label: 'cplusplushater@icloud.com' },
+    //     { value: 'email3', label: 'ISEBestBuilding@yahoo.com' },
+    // ];
+
+    // const programOptions = [
+    //     { value: 'youthProgram', label: 'Youth Program' },
+    //     { value: 'retailSales', label: 'Retail Sales' },
+    //     { value: 'recycle', label: 'Recycle' },
+    //     { value: 'earnABicycle', label: 'Earn-a-bicycle' },
+    //     { value: 'earnAComputer', label: 'Earn-a-computer' },
+    // ];
 
-    const donorEmailOptions = [
-        { value: 'email1', label: 'cooldude@gmail.com' },
-        { value: 'email2', label: 'cplusplushater@icloud.com' },
-        { value: 'email3', label: 'ISEBestBuilding@yahoo.com' },
-    ];
+    const [donorEmailOptions, setDonorEmailOptions] = useState<Option[]>([]);
+    const [programOptions, setProgramOptions] = useState<Option[]>([]);
+
+    useEffect(() => {
+        //fetching the donor emails
+        const fetchDonorEmails = async () => {
+            try {
+                const response = await axios.get('http://localhost:4000/donor');
+                const emailOptions = response.data.map((donor: any) => ({
+                    value: donor.email,
+                    label: donor.email,
+                    id: donor.id, // Store the donor ID
+                }));
+                setDonorEmailOptions(emailOptions);
+            } catch (error) {
+                console.error('Error fetching donor emails', error);
+            }
+        };
 
-    const programOptions = [
-        { value: 'youthProgram', label: 'Youth Program' },
-        { value: 'retailSales', label: 'Retail Sales' },
-        { value: 'recycle', label: 'Recycle' },
-        { value: 'earnABicycle', label: 'Earn-a-bicycle' },
-        { value: 'earnAComputer', label: 'Earn-a-computer' },
-    ];
+        const fetchPrograms = async () => {
+            try {
+                const response = await axios.get(
+                    'http://localhost:4000/program',
+                );
+                const programOptions = response.data.map((program: any) => ({
+                    value: program.name,
+                    label: program.name,
+                    id: program.id, // Store the donor ID
```

Comment:
It requires only `value` and `label` for `select menu`, remove the `id` from here. And the `value` here should hold the `program id`.  Same applies to `emailOptions` above